packages/components/src/styles/components/modal.scss

```diff
@@ -8,16 +8,18 @@
 //
 
 .hds-modal {
+  position: fixed;
   z-index: 50;
-  flex-direction: column;
+  height: fit-content;
+  margin: auto;
   padding: 0;
   background: var(--token-color-surface-primary);
   border: none;
   border-radius: 8px;
   box-shadow: var(--token-surface-overlay-box-shadow);
```

Comment:
reminder: let's abstract also the `Dialog::Overlay` component, it's shared in both the `Modal` and the `Flyout`